desktop/cmp/input/DateInput.ts

```diff
@@ -71,7 +71,7 @@ export interface DateInputProps extends HoistProps, LayoutProps, HoistInputProps
      * Element to display inline on the right side of the input. Note if provided, this will
      * take the place of the (default) calendar-picker button and (optional) clear button.
      */
-    rightElement?: ReactNode;
+    rightElement?: JSX.Element;
```

Comment:
?

Reply:
Changed since this gets passed to BP `InputGroup` which accepts `rightElement?: JSX.Element` (more constrained type than `ReactNode`)

Reply:
is this a typescript error with BP's signature, or a runtime error?  If ReactNodes generally work fine, than we should just cast internally.

Don't want to limit ourselves based on BP, and typescript any more than neccessary

Reply:
I just tested `string`, `number`, `boolean` and `Iterable<ReactNode>` which are all valid `ReactNodes` but not `JSX.Elements`.  All work without causing runtime exceptions.

I hadn't noticed this before, but our type for `leftIcon` is `ReactElement`, whereas BP's is `IconName | MaybeElement`.  `IconName` is a union of supported string literals, and `MaybeElement = JSX.Element | false | null | undefined`.  It seems like if we're going to deviate from BP's types, these ought to be the same?